cdap-data-fabric/src/main/java/co/cask/cdap/store/OwnerStore.java

```diff
@@ -0,0 +1,207 @@
+ * the License.
+ */
+
+package co.cask.cdap.store;
+
+import co.cask.cdap.api.Transactional;
+import co.cask.cdap.api.TxRunnable;
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.data.DatasetContext;
+import co.cask.cdap.api.dataset.DatasetManagementException;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.table.ConflictDetection;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.common.AlreadyExistsException;
+import co.cask.cdap.common.kerberos.SecurityUtil;
+import co.cask.cdap.data.dataset.SystemDatasetInstantiator;
+import co.cask.cdap.data2.datafabric.dataset.DatasetsUtil;
+import co.cask.cdap.data2.dataset2.DatasetFramework;
+import co.cask.cdap.data2.dataset2.MultiThreadDatasetCache;
+import co.cask.cdap.data2.transaction.Transactions;
+import co.cask.cdap.data2.transaction.TxCallable;
+import co.cask.cdap.proto.id.DatasetId;
+import co.cask.cdap.proto.id.EntityId;
+import co.cask.cdap.proto.id.NamespaceId;
+import com.google.inject.Inject;
+import org.apache.tephra.RetryStrategies;
+import org.apache.tephra.TransactionFailureException;
+import org.apache.tephra.TransactionSystemClient;
+
+import java.io.IOException;
+import java.util.Collections;
+
+/**
+ * This class manages owner's principal information of CDAP entities.
+ * <p>
+ * Currently: Owner information is stored for the following entities:
+ * <ul>
+ * <li>{@link co.cask.cdap.api.data.stream.Stream}</li>
+ * <li>{@link co.cask.cdap.api.dataset.Dataset}</li>
+ * <li>{@link co.cask.cdap.api.app.Application}</li>
+ * </ul>
+ * </p>
+ * <p>
+ * It is the responsibility of the creator of the supported entities to add an entry in this store to store the
+ * associated owner's principal. Note: An absence of an entry in this table for an {@link EntityId} does not
+ * signifies that the entity does not exists. The owner information is only stored if an owner was provided during
+ * creation time else the owner information is non-existent which signifies that the entity own is default CDAP owner.
+ * </p>
+ * <p>
+ * <p>
+ * The owner's principal must be in the of the following Kerberos name types:
+ * <ul>
+ * <li>KRB_NT_PRINCIPAL:  Just the name of the principal as in DCE, or for users. For example: alice@REALM</li>
+ * <li>KRB_NT_SRV_HST:  Service with host name as instance(telnet, rcommands). For example alice/hostname@REALM</li>
+ * </ul>
+ * Refer to <a href=https://tools.ietf.org/html/rfc4120#section-7.5.8>Name Types</a> documentation for details on
+ * Kerberos Name Types.
+ * </p>
+ */
+public class OwnerStore {
+  private static final String OWNER_PREFIX = "o";
+  // currently, we only leverage one column of the table. However, not using KeyValueTable, so that being able to use
+  // additional columns in the future is simple
+  private static final byte[] COL = Bytes.toBytes("c");
+  private static final DatasetId DATASET_ID = NamespaceId.SYSTEM.dataset("owner.meta");
```

Comment:
Not sure about that, at the very least we will need to add group to this. Let's give it some time tomorrow morning and if we can't come up with something better than we can go with the current name.